tests/test_pv_unittests.py

```diff
@@ -299,6 +299,25 @@ def test_put_string_waveform_zero_length_strings():
         get_value = pv.get(use_monitor=False, as_numpy=False)
         numpy.testing.assert_array_equal(get_value, put_value)
 
+
+def test_string_waveform_lengths():
+    "test string arrays with length=1 and other lengths"
+    spv = get_pv(pvnames.string_arr_pv)
+    readback = spv.get()
+    nelm = spv.nelm
+    with no_simulator_updates():
+        n = 0
+        for count in (nelm, nelm-2, nelm//3, 11, 8, 5, 3, 2, 1):
+            n += 5*count
+            dat = [f"str_{i+n}" for i in range(count)]
+            spv.put(dat)
+
+            time.sleep(0.1)
+            readback = spv.get()
+            assert all(readback == dat)
+            #            print(count, dat, readback)
```

Comment:
Assuming pytest is used, the print and the assert can be merged into one using:
assert readback == dat, f"Count {count}"